injection/sharedmain/main.go

```diff
@@ -120,12 +119,14 @@ var (
 // by name.
 func MainNamed(ctx context.Context, component string, ctors ...injection.NamedControllerConstructor) {
 
-	disabledControllers := pflag.StringSlice("disable-controllers", []string{}, "Comma-separated list of disabled controllers.")
-
+	var (
+		disabledControllers string
+	)
+	flag.StringVar(&disabledControllers, "disable-controllers", "", "Comma-separated list of disabled controllers.")
 	// HACK: This parses flags, so the above should be set once this runs.
 	cfg := injection.ParseAndGetRESTConfigOrDie()
 
-	enabledCtors := enabledControllers(*disabledControllers, ctors)
+	enabledCtors := enabledControllers(strings.Split(disabledControllers, ","), ctors)
```

Comment:
Im not sure I understand why we want to do the split ourselves here rather than using the pflag.StringSlice library which seems more concise / avoids the extra local var, and is a perfectly valid flag.Var?

Reply:
pflag used only for this flag doesn't work.

Reply:
yeah, was not working, therefore changing :man_shrugging:

Reply:
I don't follow, isn't pflag just a standard flag.Var so should work fine with flag library? 🤔 Is there a failing test or something for this

Reply:
(I'm probably missing something obvious about flag/pflag, but I'd love to understand before we merge it why this doesn't work)

Reply:
Maybe we need this import everywhere we use `flag`?
```golang
import flag "github.com/spf13/pflag"
```

Reply:
ahh I looked at the code - seems like pflag.* adds the var to its own flag set, not the one package flag uses, and it doesn't expose the internal flag.Var externally afaict, annoyingly